lms/static/scripts/frontend_apps/components/SubmitGradeForm.js

```diff
@@ -20,18 +20,46 @@ import ValidationMessage from './ValidationMessage';
 export default function SubmitGradeForm({ disabled = false, student }) {
   const [showError, setShowError] = useState(false); // Is there a validation error message to show?
   const [gradeErrorMessage, setGradeErrorMessage] = useState(''); // The actual validation error message.
+  const [grade, setGrade] = useState('');
 
-  const [networkError, setNetworkError] = useState(''); // if there is an error when submitting grade
-  const [requestStatus, setRequestStatus] = useState(''); // ajax request state, one of ('', 'fetching', 'error', 'done')
+  const [networkError, setNetworkError] = useState(''); // If there is an error when submitting grade
+  const [requestStatus, setRequestStatus] = useState(''); // Ajax request state, one of ('', 'fetching', 'error', 'done')
+  const [fetchStatus, setFetchStatus] = useState(''); // Ajax request state, one of ('', 'fetching', 'error', 'done')
 
   const { authToken } = useContext(Config);
 
+  // Used to handle keyboard input effects.
+  const refInput = useRef(null);
+
   useEffect(() => {
     setRequestStatus(''); // clear out network status
-  }, [student]);
 
-  // Used to handle keyboard input effects.
-  const refInput = useRef(null);
+    // Fetch the grade from the service api
+    let fetchedGradeDone = false; // flag to prevent updating more than once
+    if (Object.entries(student).length) {
+      // See https://github.com/facebook/react/issues/14326 for async in useEffect
+      (async () => {
+        try {
+          setFetchStatus('fetching');
+          setGrade(''); // Clear the older grade so we don't show the wrong grade with the new student
+          const response = await fetchGrade({ student, authToken });
+          setFetchStatus('done');
+          if (!fetchedGradeDone) {
+            // Ignore if we started fetching something else
+            setGrade(response.currentScore * 10); // Scale the grade to be [0-10], grade is saved as a value between [0-1]
+          }
+        } catch (e) {
+          setNetworkError(e.errorMessage ? e.errorMessage : 'Unknown error');
+          setFetchStatus('error');
+        }
+      })();
+    } else {
+      setGrade(''); // No valid student, clear grade
+    }
+    return () => {
```

Comment:
Can you explain why the return value is a function object? I feel like I should know but I'm just not grokking it.

Reply:
explained here, but I'll give this a closer look.

https://www.robinwieruch.de/react-hooks-fetch-data